src/ert/storage/local_ensemble.py

```diff
@@ -876,3 +900,127 @@ def get_response_state(
+                            .cast(polars.Categorical)
+                            .alias("response_key")
+                        ]
+                    )
+
+                    # Filter out responses without observations
+                    for col, observed_values in observed_cols.items():
+                        responses = responses.filter(
+                            polars.col(col).is_in(observed_values)
+                        )
+
+                    pivoted = responses.collect().pivot(
+                        on="realization",
+                        index=["response_key", *response_cls.primary_key],
+                        values="values",
+                        aggregate_function="mean",
+                    )
+
+                    if pivoted.is_empty():
+                        joined = observations_for_type.with_columns(
+                            polars.Series(
+                                str(real),
+                                [np.nan] * len(observations_for_type),
+                                dtype=polars.Float32,
+                            )
+                        )
+                    elif "time" in pivoted:
+                        joined = observations_for_type.join_asof(
+                            pivoted,
+                            by=["response_key", *response_cls.primary_key],
+                            on="time",
+                            tolerance="1s",
+                        )
+                    else:
+                        joined = observations_for_type.join(
+                            pivoted,
+                            how="left",
+                            on=["response_key", *response_cls.primary_key],
+                        )
+
+                    joined = (
+                        joined.with_columns(
+                            polars.concat_str(
+                                response_cls.primary_key, separator=", "
+                            ).alias(
+                                "__tmp_index_key__"
+                                # Avoid potential collisions w/ primary key
+                            )
+                        )
+                        .drop(response_cls.primary_key)
+                        .rename({"__tmp_index_key__": "index"})
+                    )
+
+                    # if not joined.is_empty():
+                    if first_columns is None:
+                        first_columns = joined.select(
+                            [
+                                "response_key",
+                                "index",
+                                "observation_key",
+                                "observations",
+                                "std",
+                                str(real),
+                            ]
+                        )
+                    else:
+                        realization_columns.append(joined.select(str(real)))
+
+                if first_columns is not None:
```

Comment:
`else` to the `if` above?

Reply:
Pertains to the current one. They both add a column to the dataset, only the first-occurrence adds the index columns while the following occurrence adds only a col of responses

Reply:
So this is for the case of no `reals`? Might be worth writing that explicitly before the `for` loop, a lot of changing variables to keep track of otherwise.